tests/director/tests/bundle_instance_auth_per_scenario_test.go

```diff
@@ -0,0 +1,290 @@
+	ctx := context.Background()
+	createScenarioLabelDefIfNotExist(t, ctx)
+
+	commonScenario := "common-scenario"
+	otherAppScenario := "other-app-scenario"
+	otherRuntimeScenario := "other-runtime-scenario"
+
+	scenarios := []string{conf.DefaultScenario, commonScenario, otherAppScenario, otherRuntimeScenario}
+	fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), scenarios)
+	defer fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), scenarios[:1])
+
+	// GIVEN
+	data, cleanup, ok := setupAppAndRuntimeWithCommonScenario(t, ctx, commonScenario)
+	defer cleanup()
+	require.True(t, ok)
+
+	t.Run("Remove Application from scenario by setApplicationLabel mutation", func(t *testing.T) {
+		request := fixtures.FixSetApplicationLabelRequest(data.appId, ScenariosLabel, []interface{}{otherAppScenario})
+		label := graphql.Label{}
+		err := testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &label)
+		require.Error(t, err)
+	})
+
+	t.Run("Remove Runtime from scenario by setRuntimeLabel mutation", func(t *testing.T) {
+		request := fixtures.FixSetRuntimeLabelRequest(data.runtimeId, ScenariosLabel, []interface{}{otherRuntimeScenario})
+		label := graphql.Label{}
+		err := testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &label)
+		require.Error(t, err)
+	})
+
+	t.Run("Remove Application from scenario by deleteApplicationLabel mutation", func(t *testing.T) {
+		request := fixtures.FixDeleteApplicationLabelRequest(data.appId, ScenariosLabel)
+		label := graphql.Label{}
+		err := testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &label)
+		require.Error(t, err)
+	})
+
+	t.Run("Remove Runtime from scenario by deleteRuntimeLabel mutation", func(t *testing.T) {
+		request := fixtures.FixDeleteRuntimeLabelRequest(data.runtimeId, ScenariosLabel)
+		label := graphql.Label{}
+		err := testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &label)
+		require.Error(t, err)
+	})
+
+	t.Run("Remove Runtime from scenario by updateRuntime mutation", func(t *testing.T) {
+		updateInput := graphql.RuntimeInput{
+			Name:   data.runtimeName,
+			Labels: graphql.Labels{ScenariosLabel: "other-runtime-scenario"},
+		}
+		updateInputGQL, err := testctx.Tc.Graphqlizer.RuntimeInputToGQL(updateInput)
+		require.NoError(t, err)
+
+		request := fixtures.FixUpdateRuntimeRequest(data.runtimeId, updateInputGQL)
+		input := graphql.RuntimeExt{}
+		err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &input)
+		require.Error(t, err)
+	})
+}
+
+func TestUnassignResourceFromScenarioUsingAutomaticScenarioAssignment(t *testing.T) {
+	ctx := context.Background()
+	tenantID := tenant.TestTenants.GetDefaultTenantID()
+	createScenarioLabelDefIfNotExist(t, ctx)
+
+	var cleanups []func()
+	testCleanUp := func() {
+		for _, cleanup := range cleanups {
+			cleanup()
+		}
+	}
+	defer testCleanUp()
```

Comment:
Those are not needed you can just defer all the clean-ups in place instead of storing them in a slice since you are in a top-level test already.